components/AppBar/src/MDCAppBar.m

```diff
@@ -47,7 +47,13 @@ - (void)mutate:(nonnull MDCAppBar *)appBar {
   UIColor *backgroundColor = appBar.navigationBar.backgroundColor ?:
       appBar.headerViewController.headerView.backgroundColor;
   if (!backgroundColor) {
-    return;
+    // It might be possible that the views haven't received their background colors if we're using
+    // UIAppearance to style them. That's why we check their appearance before giving up.
+    backgroundColor = [[MDCNavigationBar appearance] backgroundColor] ?:
```

Comment:
I have strong reservations about this change. It's possible that the color was explicitly set to `nil`, and that won't be caught by this check.  UIAppearance won't apply the proxied values to an instance if the setter was called.   This may be the case for an AppBar with an imageView as its background (instead of a color), or one that relies on the `backgroundColor` of the container view.

If we **really** want to take this approach, please update the public (header) documentation to make it clear that it will overwrite any instance values with UIAppearance values (if set).  My vote is to update our examples to do mutating in `-viewDidAppear:` and provide that as the best practice.

Reply:
`viewDidAppear` is not a realistic option. Other ideas?

Reply:
Did you explore running the mutator on the UIAppearance instance?

Reply:
No, would that be considered a better approach?

Reply:
IMO it would. That way clients using UIAppearance can just use directly after configuring their UIAppearance setup, and clients manually setting values on an MDCAppBar instance can mutate their instance.

Reply:
FWIW, if you're setting your background color to nil you should not be calling this mutator as it only works if you do have a background color.

Reply:
Thoughts on @samnm's proposal @romoore @ianegordon?

Reply:
I think having the client call it explicitly on UIAppearance proxy is better. If it's not already documented, we should ensure they realize they'll have to theme both the FHV and NavBar proxies before calling the mutator on the AppBar proxy.

Reply:
Since MDCAppBar is not a view, I can't just use UIAppearance. I'll just drop this PR.

Reply:
Applying the mutator to the UIAppearance proxy seemed like such a clean solution.  :/
Since MDCAppBar is not a UIView would a possible solution be class properties similar to what we had to do for textfields?